sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileStatusCacheSensorTest.java

```diff
@@ -0,0 +1,146 @@
+    sensor.describe(sensorDescriptor);
+
+    assertThat(sensorDescriptor.name()).isEqualTo("File status cache sensor");
+    assertThat(sensorDescriptor.languages()).isEmpty();
+  }
+
+  @Test
+  public void execute_whenAnalyzingAPullRequest_logsMessage() throws IOException {
+    var settings = new MapSettings().setProperty("sonar.pullrequest.key", "42");
+    var context = SensorContextTester.create(temp.newFolder()).setSettings(settings);
+    var sensor = new FileStatusCacheSensor(new HashProvider());
+
+    sensor.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Incremental PR analysis: Cache is not updated for pull requests.");
+  }
+
+  @Test
+  public void execute_whenPullRequestCacheBasePathIsNotConfigured_logsWarning() throws IOException {
+    var context = SensorContextTester.create(temp.newFolder());
+    var sut = new FileStatusCacheSensor(new HashProvider());
+
+    sut.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Incremental PR analysis: Pull request cache base path is not configured. Skipping file status upload.");
+  }
+
+  @Test
+  public void execute_whenCacheIsDisabled_logsWarning() throws IOException {
+    var settings = new MapSettings().setProperty("sonar.pullrequest.cache.basepath", "C:/");
+    var context = SensorContextTester.create(temp.newFolder()).setSettings(settings);
+    context.setCacheEnabled(false);
+    var sensor = new FileStatusCacheSensor(new HashProvider());
+
+    sensor.execute(context);
+
+    assertThat(logTester.logs()).containsExactly("Incremental PR analysis: Analysis cache is disabled. Skipping file status upload for incremental PR analysis.");
+  }
+
+  @Test
+  public void execute_whenCacheIsEnabled_itAddsTheFiles() throws IOException, NoSuchAlgorithmException {
+    var hashProvider = mock(HashProvider.class);
+    when(hashProvider.computeHash(any())).thenReturn(new byte[0]);
+    var context = CreateContextForCaching();
+    var sut = new FileStatusCacheSensor(hashProvider);
+
+    sut.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly(
+      "Incremental PR analysis: Preparing to upload file status.",
+      "Incremental PR analysis: Adding VB/Bar.vb hash to cache.",
+      "Incremental PR analysis: Adding CSharp/Foo.cs hash to cache."
+    );
+  }
+
+  @Test
+  public void execute_whenHashingFails_itLogsAnError() throws IOException, NoSuchAlgorithmException {
+    var hashProvider = mock(HashProvider.class);
+    when(hashProvider.computeHash(any())).thenThrow(new IOException());
+    var context = CreateContextForCaching();
+    var sut = new FileStatusCacheSensor(hashProvider);
+
+    sut.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactly(
+      "Incremental PR analysis: An error occurred while computing the hash for VB/Bar.vb",
```

Comment:
Can we log at least message of the error? "There's an error" isn't very helpful per say

Reply:
The exceptions are logged and displayed in the console. They are not asserted since the `logTester` returns the messages like this.
We can assert by index access but I'm not sure if it's worth it.